### PR TITLE
[FIX] Fixed `ModalContainer` not working in `StrictMode`

### PR DESCRIPTION
**Pull request by razvantomegea**
### Issue
Modals do not appear in the Ledger and WalletConnect login buttons when using `StrictMode`

### Reproduce
Add `StrictMode` around the main `App` component and try to login using `LedgerLoginButton` or `WalletConnectButton`

### Root cause
The `useDappModal` manages the state in redux store with 2 effect hooks that alter the visibility of the modal.

### Fix
Remove modal management from redux and handle the configs locally.

### Additional changes
Redirect with page refresh after login with `window.location.assign`, instead of just `history.pushState`. The latter was not working well.

### Contains breaking changes
[x] No

[] Yes

### Updated CHANGELOG
[x] Yes

### Testing
[x] User testing
[x] Unit tests

### CODE REVIEW

File: src/UI/DappModal/dappModal.types.ts [hunk 1]
```diff
@@ -15,6 +15,5 @@ export interface DappModalConfig {
 }
 
 export interface DappModalInteractionConfig {
-  openOnMount?: boolean;
```
Comment (arhtudormorar) [inline src/UI/DappModal/dappModal.types.ts:18]: Is this not breaking?

File: src/UI/DappModal/hooks/useDappModal.ts [hunk 2]
```diff
@@ -1,71 +1,29 @@
...
 
-export interface UseDappModalProps {
-  config?: DappModalInteractionConfig;
-  modalConfig?: DappModalConfig;
-}
+import { DappModalInteractionConfig } from '../dappModal.types';
 
-export const useDappModal = (props?: UseDappModalProps) => {
-  const dispatch = useDispatch();
-  const visible = useIsDappModalVisible();
```
Comment (arhtudormorar) [inline src/UI/DappModal/hooks/useDappModal.ts:21]: I think this was persisting the visibility in some cases

File: src/utils/logout.ts [hunk 3]
```diff
@@ -89,7 +89,7 @@ function redirectToCallbackUrl(
     if (typeof onRedirect === 'function') {
       onRedirect(callbackUrl);
     } else {
-      safeRedirect({ url: callbackUrl, shouldForcePageReload: true });
```
Comment (arhtudormorar) [inline src/utils/logout.ts:92]: why this change?

File: src/utils/redirect.ts [hunk 4]
```diff
@@ -19,13 +17,6 @@ export const safeRedirect = ({
         return;
       }
 
-      // Navigate to callbackUrl without page refresh if we are in the same origin
```
Comment (arhtudormorar) [inline src/utils/redirect.ts:22]: It would have been better if this was in a separate PR

File: src/UI/DappModal/hooks/useDappModal.ts (hunk 2 above)
Comment (razvantomegea) [reply src/UI/DappModal/hooks/useDappModal.ts:21]: It is not necessary to have one centralised modal managed with redux

File: src/UI/DappModal/dappModal.types.ts (hunk 1 above)
Comment (razvantomegea) [reply src/UI/DappModal/dappModal.types.ts:18]: No, because we have the `visible` props/flag which was already doing that

File: src/utils/logout.ts (hunk 3 above)
Comment (razvantomegea) [reply src/utils/logout.ts:92]: Will make another PR for this

File: src/utils/redirect.ts (hunk 4 above)
Comment (razvantomegea) [reply src/utils/redirect.ts:22]: Will make another PR for this

File: src/utils/logout.ts (hunk 3 above)
Comment (razvantomegea) [reply src/utils/logout.ts:92]: https://github.com/multiversx/mx-sdk-dapp/pull/944

File: src/utils/redirect.ts (hunk 4 above)
Comment (razvantomegea) [reply src/utils/redirect.ts:22]: https://github.com/multiversx/mx-sdk-dapp/pull/944

File: src/UI/DappModal/hooks/useDappModal.ts (hunk 2 above)
Comment (CiprianDraghici) [reply src/UI/DappModal/hooks/useDappModal.ts:21]: I think the idea was to have this type of modal visible once at a time

Comment (razvantomegea) [reply src/UI/DappModal/hooks/useDappModal.ts:21]: This implies the component is also rendered once, like `<SignTransactionsModals />`, not multiple times in the app

**Review by CiprianDraghici — APPROVED** [review]
